Some minor tweaks (#23150)
cloudflare/cloudflare-docs: src/content/docs/containers/get-started.mdx

Now that your environment is set up, you can write your Worker.

### Deploy from the Hello World Template

Use [C3](/learning-paths/workers/get-started/c3-and-wrangler/#c3) to create and deploy a new Worker using the "Hello World Containers" template.

```sh
npm create cloudflare@latest -- --template=cloudflare/templates/hello-world-containers
```

Subsequent updates to the Worker or Container code can be deployed using Wrangler:

<PackageManagers type="exec" pkg="wrangler" args="deploy" />

When you run this command, several things happen:

- Wrangler builds your image using Docker.
- Wrangler pushes your image to the Cloudchamber Image Registry, which is automatically
  integrated with your Cloudflare account.
- Wrangler deploys your Worker, which includes setting up a "Container" for it to interact with.

:::note
The build and push usually take the longest on the first deploy. Future deploys
will go faster by [reusing cached image layers](https://docs.docker.com/build/cache/).
:::

After you deploy your Worker for the first time, you will need to wait several minutes until
it is ready to receive requests. Unlike Workers, Containers take a few minutes to be provisioned.

> During this time, requests are sent to the Worker, but calls to the Container will error.

Author: yeah - can swap

```js
export class MyContainer extends Container {
  defaultPort = 8080;
  sleepAfter = '10s';
  envVars = {
    MESSAGE: 'I was passed in via the container class!',
  };

  override onStart() {
    console.log('Container successfully started');
  }

  override onStop() {
    console.log('Container successfully shut down');
  }

  override onError(error: unknown) {
    console.log('Container error:', error);
  }
}
```

This defines basic configuration for the container:

- `defaultPort` sets the port that the `fetch` and `containerFetch` methods will use to communicate with the container. It also blocks
  requests until the container is listening on this port.
- `sleepAfter` sets the timeout for the container to sleep after it has been idle for a certain amount of time.
- `envVars` sets environment variables that will be passed to the container when it starts.
- `onStart`, `onStop`, and `onError` are hooks that run when the container starts, stops, or errors, respectively.

See the [Container class documentation](/containers/container-package) for more details and configuration options.

#### Routing to Containers

> In the Worker's main `fetch` handler, Containers are launched in one of two ways:

Reviewer: ```suggestion
When a request enters Cloudflare, your Worker's [`fetch` handler](/workers/runtime-apis/handlers/fetch/) is invoked. This is the code that handles the incoming request. The fetch handler in the example code, launches containers in two ways, on different routes:
```

- Making requests to `/container/` passes requests to a new container for

>   each path. This is done by spinning up a new Container instance. You may note

Reviewer: > for each path

But the example below always uses the same name — `"container"` that gets passed into `idFromName`?

  that the first request to a new path takes longer than subsequent requests, this is
  because a new container is booting.

  ```js
  if (pathname.startsWith("/container")) {

>   	let id = env.MY_CONTAINER.idFromName("container");

Reviewer: Realizing that nothing in this page or elsewhere in docs links to or explains bindings

  	let container = env.MY_CONTAINER.get(id);
  	return await container.fetch(request);
  }
  ```

- Making requests to `/lb` will load balance requests across several containers.

>   This uses a simple `getRandom` helper method currently, which picks an ID at random

Reviewer: ```suggestion
  This uses a simple `getRandom` helper method, which picks an ID at random
```

>   from a set number (in this case 3), then routes to that Container instance.

Reviewer: ```suggestion
  from a set number (in this case 3), then routes to that Container instance. You can replace this with any routing or load balancing logic you choose to implement:
```

(or maybe there is a good example we want to point towards)

  ```js
  if (pathname.startsWith("/lb")) {
  	let container = await getRandom(env.MY_CONTAINER, 3);
  	return await container.fetch(request);
  }
  ```

This allows for multiple ways of using Containers:

- If you simply want to send requests to many stateless and interchangeable containers,
  you should load balance.
- If you have stateful services or need individually addressable
  containers, you should request specific Container instances.
- If you are running short-lived jobs, want fine-grained control over the container
  lifecycle, want to parameterize container entrypoint or env vars, or
  want to chain together multiple container calls, you should request specific
  Container instances.

:::note
Currently, routing requests to one of many interchangeable Container instances is accomplished
with the `getRandom` helper.

> This is a temporary solution until first-party support for utilization-aware autoscaling

Reviewer: ```suggestion
This is temporary — we plan to add native support for latency-aware autoscaling and load balancing in the coming months.
```

> and latency-aware load balancing is added to Cloudflare Containers. We plan to add this

Reviewer: ```suggestion
```

> in the coming months.

Reviewer: ```suggestion
```

- "Containers: Edit"
- "Workers Scripts: Edit"

Refer to [Find zone and account IDs](/fundamentals/setup/find-account-and-zone-ids/) for help locating
your Account ID and [Create an API token](/fundamentals/api/get-started/create-token/) to learn how to create
an access your API token.

### Ensure Docker is running locally

In this guide, we will build and push a container image alongside your Worker code. By default, this process uses
Docker to do so. You must have Docker running locally when you run `wrangler deploy`.

If Docker is running properly, the `docker info` command will successfully run. If Docker is not running,
the `docker info` command will hang or return an error including the message "Cannot connect to the Docker daemon".

{/* FUTURE CHANGE: Add some image you can use if you don't have Docker running. */}
{/* FUTURE CHANGE: Link to docs on alternative build/push options */}

## Deploy a Container to Workers

Now that your environment is set up, you can write your Worker.

### Deploy from the Hello World Template

Use [C3](/learning-paths/workers/get-started/c3-and-wrangler/#c3) to create and deploy a new Worker using the "Hello World Containers" template.

> ```sh

Reviewer 2: [PackageManagers](https://developers.cloudflare.com/style-guide/components/package-managers/) would be a better option
